activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/types/VariableType.java

```diff
@@ -16,15 +16,16 @@
 package org.activiti.spring.process.variable.types;
 
 import java.util.List;
-
+import java.util.Objects;
+import java.util.regex.Pattern;
 import org.activiti.engine.ActivitiException;
 
 /**
- * Base variable type for types as defined in extension json files.
- * Used to validate variables against definition.
+ * Base variable type for types as defined in extension json files. Used to validate variables against definition.
  */
 public abstract class VariableType {
 
+    private static String expressionPattern = "^\\$\\{(.*)[\\}]$";
```

Comment:
```suggestion
    private static String expressionPattern = "^\\$\\{(.|\\n)*[\\}]$";
```
If we want to support multiline expressions, then we should change the pattern like this

Reply:
Use here a compiled pattern instead
```suggestion
    private static final Pattern EXPRESSION_PATTERN = Pattern.compile("^\\$\\{(.|\\n)*[\\}]$");
```

Reply:
Done